modules/bf2.py

```diff
@@ -189,72 +190,74 @@ def update_state(self, server, oldstate, newstate):
+
             squadname = self.id_to_squad_name[npi["squad"]]
-            
+
             # Add to game group
             location = "base"
             group = "bf2_%s_game" % (ng or ngcfgname)
             server.addUserToGroup(ngcfg[location], session, group)
             log.debug("Added '%s' @ %s to group %s in %s", newstate.name, ng or ngcfgname, group, location)
-            
+
             # Then add to team group
             location = npi["team"]
             group = "bf2_team"
             server.addUserToGroup(ngcfg[location], session, group)
             log.debug("Added '%s' @ %s to group %s in %s", newstate.name, ng or ngcfgname, group, location)
-            
+
             # Then add to squad group
             group = "bf2_%s_squad" % squadname
             server.addUserToGroup(ngcfg[location], session, group)
             log.debug("Added '%s' @ %s to group %s in %s", newstate.name, ng or ngcfgname, group, location)
-            
+
             channame = "%s_%s_squad" % (npi["team"], self.id_to_squad_name[npi["squad"]])
             newstate.channel = ngcfg[channame]
-            
+
             if npi["squad_leader"]:
                 # In case the leader flag is set add to leader group
                 group = "bf2_%s_squad_leader" % squadname
                 server.addUserToGroup(ngcfg[location], session, group)
                 log.debug("Added '%s' @ %s to group %s in %s", newstate.name, ng or ngcfgname, group, location)
-                
+
                 group = "bf2_squad_leader"
                 server.addUserToGroup(ngcfg[location], session, group)
                 log.debug("Added '%s' @ %s to group %s in %s", newstate.name, ng or ngcfgname, group, location)
-                
+
                 # Override previous moves
                 channame = "%s_%s_squad_leader" % (npi["team"], self.id_to_squad_name[npi["squad"]])
                 newstate.channel = ngcfg[channame]
-            
+
             if npi["commander"]:
                 group = "bf2_commander"
                 server.addUserToGroup(ngcfg[location], session, group)
                 log.debug("Added '%s' @ %s to group %s in %s", newstate.name, ng or ngcfgname, group, location)
-                
+
                 # Override previous moves
                 channame = "%s_commander" % npi["team"]
                 newstate.channel = ngcfg[channame]
-                
+
         if oli and not nli:
-            log.debug("User '%s' (%d|%d) on server %d no longer linked", newstate.name, newstate.session, newstate.userid, sid)
+            log.debug("User '%s' (%d|%d) on server %d no longer linked", newstate.name, newstate.session,
+                      newstate.userid, sid)
             server.removeUserFromGroup(0, session, "bf2_linked")
-                
-        if newstate.channel >= 0 and newoldchannel != newstate.channel:
-            if ng == None:
+
+        if 0 <= newstate.channel != newoldchannel:
```

Comment:
thought: I find this double-sided comparison confusing; not very readable or clear.
One condition checks general validity of the value to be in the value range. The second checks for value change for this particular use case/section. So it mixes two concerns too.

Reply:
I agree this is somewhat confusing, but the logic is actually the same. Most likely the whole if should be rewritten to make it more readable.